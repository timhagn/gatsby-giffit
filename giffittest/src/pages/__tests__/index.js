import React from 'react'
import { render } from '@testing-library/react'
import { StaticQuery, useStaticQuery } from 'gatsby'
import IndexPage from '..'

// Necessary to fixate generated className.
jest.mock('short-uuid')

const fluidMock = {
  childImageSharp: {
    fluid: {
      aspectRatio: 1,
      sizes: `100 200 300`,
      src: `pretend-i-am-a-base64-encoded-image`,
      srcSet: `asdfasdf`,
    },
  },
}

const mockMetadata = {
  site: {
    siteMetadata: {
      title: `gbitest`,
      description: `A simple starter to show what gatsby-background-image can do.`,
      author: `Tim Hagn`,
    },
  },
}

describe('IndexPage', () => {
  beforeEach(() => {
    StaticQuery.mockImplementation(({ render }) => render({ ...mockMetadata }))
    useStaticQuery.mockImplementation(() => ({
      desktop: fluidMock,
      placeholderImage: fluidMock,
      ...mockMetadata,
    }))
    // Freeze generated className.
    const uuid = require('short-uuid')
    uuid.generate.mockImplementation(() => '73WakrfVbNJBaAmhQtEeDv')
  })

  it('renders correctly', () => {
    const { container } = render(<IndexPage />)
    expect(container).toMatchSnapshot()
  })
})
