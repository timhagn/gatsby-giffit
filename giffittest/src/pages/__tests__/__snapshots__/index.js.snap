// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexPage renders correctly 1`] = `
<div>
  <header
    style="background: rgb(0, 68, 111); position: fixed; top: 0px; left: 0px; z-index: 1; width: 100vw;"
  >
    <div
      style="margin: 0px auto; padding: 1.45rem 1.0875rem;"
    >
      <h1
        style="margin: 0px; text-align: center;"
      >
        <a
          href="/"
          style="color: white; text-decoration: none;"
        >
          gbitest
        </a>
      </h1>
    </div>
    <nav
      class="sc-gKsewC ldZnQy"
    >
      <a
        class="sc-jSgupP sc-iBPRYJ gmnWVn guyYKo"
        href="/"
        nav="/"
      >
        Home
      </a>
      <a
        class="sc-jSgupP sc-iBPRYJ gmnWVn laLoyD"
        href="/image-stack/"
        nav="/image-stack/"
      >
        Image Stack
      </a>
      <a
        class="sc-jSgupP sc-iBPRYJ gmnWVn laLoyD"
        href="/image-stack-2/"
        nav="/image-stack-2/"
      >
        Image Stack
      </a>
      <a
        class="sc-jSgupP sc-iBPRYJ gmnWVn laLoyD"
        href="/art-direction/"
        nav="/art-direction/"
      >
        Art Direction
      </a>
    </nav>
  </header>
  <div
    class="sc-gsTCUz jedleH"
  >
    <div
      class="sc-fubCfw letjyZ"
    >
      <section
        aria-label="gbitest"
        class="sc-jrAGrp eDrFkH gbi--802263448-undefined"
        id="gbitest"
        role="img"
        style="background-position: center; background-repeat: no-repeat; background-size: cover; position: relative; opacity: 0.99;"
        title="gbitest"
      >
        <style>
          
          .sc-jrAGrp.eDrFkH.gbi--802263448-undefined:before,
          .sc-jrAGrp.eDrFkH.gbi--802263448-undefined:after {
            content: '';
            display: block;
            position: absolute;
            width: 100%;
            height: 100%;
            top: 0;
            left: 0;
            background-color: #040e18;
            transition: opacity 0.5s ease 250ms;
            background-position: center;
background-repeat: no-repeat;
background-size: cover;

          }
          .sc-jrAGrp.eDrFkH.gbi--802263448-undefined:before {
            z-index: -100;
            
            
            opacity: 1; 
          }
          .sc-jrAGrp.eDrFkH.gbi--802263448-undefined:after {
            z-index: -101;
            
            
            
          }
        
        </style>
        <div
          class="sc-bdfBwQ hwocte"
        >
          <div
            class="sc-dlfnbm sc-hKgILt ivkxXu dblVyR"
          >
            <div>
              <h1>
                Hi people
              </h1>
              <p>
                Welcome to the 
                <code>
                  gatsby-background-image/src
                </code>
                 test site.
              </p>
              <p>
                As the left background, you see an image rendered by
                 
                <code>
                  gatsby-background-image/src
                </code>
              </p>
              <p>
                To the right, the same image rendered by 
                <code>
                  gatsby-image
                </code>
              </p>
              <div
                class="sc-eCssSg kMTTKe"
              >
                <div
                  class=" gatsby-image-wrapper"
                  style="position: relative; overflow: hidden; border-radius: 15px;"
                >
                  <div
                    aria-hidden="true"
                    style="width: 100%; padding-bottom: 100%;"
                  />
                  <picture>
                    <source
                      sizes="100 200 300"
                      srcset="asdfasdf"
                    />
                    <img
                      alt=""
                      loading="lazy"
                      sizes="100 200 300"
                      src="pretend-i-am-a-base64-encoded-image"
                      srcset="asdfasdf"
                      style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
                    />
                  </picture>
                  <noscript>
                    &lt;picture&gt;&lt;source srcset="asdfasdf" sizes="100 200 300" /&gt;&lt;img loading="lazy" sizes="100 200 300" srcset="asdfasdf" src="pretend-i-am-a-base64-encoded-image" alt="" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
                  </noscript>
                </div>
              </div>
              Go to 
              <a
                class="sc-jSgupP gmnWVn"
                href="/image-stack/"
              >
                image stack
              </a>
               to see an example of
              <strong>
                 multiple stacked background images
              </strong>
              .
              <br />
              Go to 
              <a
                class="sc-jSgupP gmnWVn"
                href="/art-direction/"
              >
                art direction
              </a>
               to see an example of
              <strong>
                 art-directed background images
              </strong>
              .
            </div>
          </div>
        </div>
      </section>
    </div>
    <div
      class="sc-fubCfw letjyZ"
    >
      <div
        class="sc-pFZIQ NhORk gatsby-image-wrapper"
        style="position: relative; overflow: hidden;"
      >
        <div
          aria-hidden="true"
          style="width: 100%; padding-bottom: 100%;"
        />
        <div
          aria-hidden="true"
          style="background-color: rgb(4, 14, 24); position: absolute; top: 0px; bottom: 0px; opacity: 1; right: 0px; left: 0px; transition-delay: 500ms;"
        />
        <picture>
          <source
            sizes="100 200 300"
            srcset="asdfasdf"
          />
          <img
            alt=""
            loading="lazy"
            sizes="100 200 300"
            src="pretend-i-am-a-base64-encoded-image"
            srcset="asdfasdf"
            style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
          />
        </picture>
        <noscript>
          &lt;picture&gt;&lt;source srcset="asdfasdf" sizes="100 200 300" /&gt;&lt;img loading="lazy" sizes="100 200 300" srcset="asdfasdf" src="pretend-i-am-a-base64-encoded-image" alt="" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
        </noscript>
      </div>
    </div>
  </div>
  <footer
    style="position: fixed; bottom: 0px; left: 0px; background: rgb(0, 68, 111); z-index: 1; width: 100vw; padding: 1rem; text-align: center; color: white;"
  >
    Â© 
    2021
    , Built with
     
    <a
      href="https://www.gatsbyjs.org"
    >
      Gatsby
    </a>
     & 
    <a
      href="https://github.com/timhagn/gatsby-background-image"
    >
      gatsby-background-image
    </a>
  </footer>
</div>
`;
